src/services/PwmService/PwmService.ts

```diff
@@ -0,0 +1,183 @@
+   */
+  async enable(PWM: PWM): Promise<string> {
+    const response = await this.PWMRequest("enable", PWM, "EdgePiRPC_PWM.SuccessMsg");
+    return (response.content as SuccessMsg).content;
+  }
+
+  /**
+   * @async Calls the PWM disable SDK method through an RPC request
+   */
+  async disable(PWM: PWM): Promise<string> {
+    const response = await this.PWMRequest("disable", PWM, "EdgePiRPC_PWM.SuccessMsg");
+    return (response.content as SuccessMsg).content;
+  }
+
+  /**
+   * @async Calls the PWM close SDK method through an RPC request
+   */
+  async close(PWM: PWM): Promise<string> {
+    const response = await this.PWMRequest("close", PWM, "EdgePiRPC_PWM.SuccessMsg");
+    return (response.content as SuccessMsg).content;
+  }
+
+  /**
+   * @async Calls the PWM frequency SDK method through an RPC request
+   */
+  async getFrequency(PWM: PWM): Promise<number> {
+    const response = await this.PWMRequest("get_frequency", PWM, "EdgePiRPC_PWM.GetFrequency");
+    return (response.content as GetFrequency).frequency;
+  }
+
+  /**
+   * @async Calls the PWM get_duty_cycle SDK method through an RPC request
+   */
+  async getDutyCycle(PWM: PWM): Promise<number> {
+    const response = await this.PWMRequest("get_duty_cycle", PWM, "EdgePiRPC_PWM.GetDutyCycle");
+    return (response.content as GetDutyCycle).dutyCycle;
+  }
+
+  /**
+   * @async Calls the PWM get_polarity SDK method through an RPC request
+   */
+  async getPolarity(PWM: PWM): Promise<Polarity> {
+    const response = await this.PWMRequest("get_polarity", PWM, "EdgePiRPC_PWM.GetPolarity");
+    return (response.content as GetPolarity).polarity;
+  }
+
+  /**
+   * @async Calls the PWM get_enabled SDK method through an RPC request
+   */
+  async getEnabled(PWM: PWM): Promise<boolean> {
+    const response = await this.PWMRequest("get_enabled", PWM, "EdgePiRPC_PWM.GetEnabled");
+    return (response.content as GetEnabled).enabled;
+  }
+
+  /**
+   * @async Calls the PWM set_config SDK method through RPC
+   */
+  async setConfig({
+    pwmNum,
+    frequency = undefined,
+    dutyCycle = undefined,
+    polarity = undefined,
+  }: PWMConfig): Promise<string> {
+    const requestType = this.serviceProtoRoot.lookupType(
+      "EdgePiRPC_PWM.Config"
+    );
+    const responseType = this.serviceProtoRoot.lookupType(
+      "EdgePiRPC_PWM.SuccessMsg"
+    );
+    const argsList = createConfigArgsList({ pwmNum, frequency, dutyCycle, polarity });
+
+    const response = await this.handlePWMRequest(
+      requestType,
+      responseType,
+      "set_config",
+      { confArg: argsList }
+    );
+    return (response.content as SuccessMsg).content;
+  }
+
+  /**
+   * @async Sends an RPC request for a specified PWM SDK method
+   */
+  private async PWMRequest(
+    methodName: string,
+    pwmNum: PWM,
+    responseTypeName: string,
+  ): Promise<serverResponse> {
+    const requestType = this.serviceProtoRoot.lookupType(
+      "EdgePiRPC_PWM.PWM"
```

Comment:
Is it possible to use constants instead for these string args?

Reply:
Yeah, I think that would work well. Updated to use constants.